pkg/kapp/resources/resources.go

```diff
@@ -37,7 +37,17 @@ const (
 	resourcesDebug = false
 )
 
-type Resources struct {
+type Resources interface {
+	All([]ResourceType, AllOpts) ([]Resource, error)
+	Delete(Resource) error
+	Exists(Resource) (bool, error)
+	Get(Resource) (Resource, error)
+	Patch(Resource, types.PatchType, []byte) (Resource, error)
+	Update(Resource) (Resource, error)
+	Create(resource Resource) (Resource, error)
+}
+
+type Impl struct {
```

Comment:
Usually, in Go, the `Impl` term is not seen as a good naming practice. Given what this struct is trying to accomplish it looks like a struct that will hold Kubernetes resources so my suggestion is that we rename this struct to `KubernetesResources`

Reply:
given the context that we are dealing with Kubernetes to me KubernetesResources is not that appealing. it's not necessary common, but plenty of places use Impl as a suffix for a default implementations. my personal choice would be to roll with ResourcesImpl until there is something more defining about this class that would give us a more specific prefix than Kubernetes.